std/typecons.d

```diff
@@ -6829,6 +6829,18 @@ mixin template Proxy(alias a)
     bool* b = Name("a") in names;
 }
 
+version(DIP1000) // excludes struct S; it's 'mixin Proxy!foo' doesn't compile with -dip1000; see else branch (-dip25)
+@system unittest
+{
+    // bug14213, using function for the payload
+    static class C
+    {
+        int foo() { return 12; }
+        mixin Proxy!foo;
+    }
+    C c = new C();
+}
```

Comment:
The duplication wouldn't have been necessary:

```d
@system unittest
{
    // bug14213, using function for the payload
    static struct S
    {
        int foo() { return 12; }
        mixin Proxy!foo;
    }
    S s;
    assert(s + 1 == 13);
    assert(s * 2 == 24);
}

version(DIP1000) {} else
@system unittest
{
    static class C
    {
        int foo() { return 12; }
        mixin Proxy!foo;
    }
    C c = new C();
}

Reply:
@wilzbach Yes, I was aware of the duplicated class C: My idea was to simply delete my original version(DIP1000) addition again, once the reason for failing struct S mixin Proxy!foo is found.

Reply:
Deleting `version(DIP1000) {} else` is even easier ;-)